pkg/router/template/plugin.go

```diff
@@ -155,7 +157,7 @@ func (p *TemplatePlugin) HandleEndpoints(eventType watch.EventType, endpoints *k
 	switch eventType {
 	case watch.Added, watch.Modified:
 		glog.V(4).Infof("Modifying endpoints for %s", key)
-		routerEndpoints := createRouterEndpoints(endpoints, !p.IncludeUDP)
+		routerEndpoints := createRouterEndpoints(endpoints, !p.IncludeUDP, service)
```

Comment:
As mentioned above I'd prefer TemplatePlugin to have LookupService and pass that method in to the createRouterEndpoints (which then only does the lookup on changes)